### ISSUE TITLE
endless recursion parsing TED_EXPORT.xsd

### ISSUE THREAD

**Issue by joaompinto** [brunato/xmlschema#88]
The problem can be reproduced with:
```
mkdir /tmp/TED 
cd /tmp/TED
wget http://publications.europa.eu/mdr/resource/eprocurement/ted/R2.0.8/publication/latest/TED_publication_R2.0.8.S04.E01_003-20180515.zip
unzip wget http://publications.europa.eu/mdr/resource/eprocurement/ted/R2.0.8/publication/latest/TED_publication_R2.0.8.S04.E01_003-20180515.zip
python3.6 -c "from xmlschema import XMLSchema; XMLSchema('TED_EXPORT.xsd')"
```
Which results in:

> RecursionError: maximum recursion depth exceeded in comparison

**Comment by brunato** [comment]
Found and fixed the problem. It was related to complexType named "btx_oth" in the file *common_prod.xsd*.

A schema components iterations stops on group's items that are global but this doesn't work with "btx_oth" because its group items are local. So the recursion doesn't stop when there are cases like that. To solve this I've added an extra check on component iteration methods of `XsdGroup` class.

I'll close this issue when the code is pushed and the next release is published.

Thanks
